packages/datetime/src/common/dateFnsFormatUtils.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { format, type Locale, parse } from "date-fns";
+
+import type { DatePickerBaseProps } from "./datePickerBaseProps";
+import { TimePrecision } from "./timePrecision";
```

Comment:
Imports are changed to reference modules from within `@blueprintjs/datetime` directly (prevents circular dependencies)
https://github.com/palantir/blueprint/blob/ca86eebdf9e7ed242d0cfd8abaf73ac376c9a5f1/packages/datetime2/src/common/dateFnsFormatUtils.ts#L19